### PR TITLE
Add maxWidth parameter for image loader to support vector images on Android.

### PR DESCRIPTION
**Pull request by xinguohenan**
We want to support svg images, so we need a width parameter sent to image loader to decide the size of the svg image.

### CODE REVIEW

**Comment by almedina-ms** [comment]
> 

Most platforms dont support svg images yet, could you move it to the android project, I think source/android/mobile could be a good place in the meantime

---
Refers to: samples/v1.1/Scenarios/Image.SvgDataUri.json:12 in ad0891a. [](commit_id = ad0891a4c5af143933f93a0cfec1fc1b1dfd4c43, deletion_comment = False)

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/IDataUriImageLoader.java [hunk 1]
```diff
@@ -6,5 +6,5 @@
 
 public interface IDataUriImageLoader
 {
-    HttpRequestResult<Bitmap> loadDataUriImage(String url, GenericImageLoaderAsync loader) throws Exception;
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/IDataUriImageLoader.java]: >HttpRequestResult<Bitmap> loadDataUriImage(String url, GenericImageLoaderAsync loader) throws Exception; [](start = 4, length = 104)

don't delete this method, let's keep both methods

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/GenericImageLoaderAsync.java [hunk 2]
```diff
@@ -24,11 +24,13 @@
     String m_imageBaseUrl;
     IOnlineImageLoader m_onlineImageLoader = null;
     IDataUriImageLoader m_dataUriImageLoader = null;
+    int m_maxWidth;
 
-    GenericImageLoaderAsync(RenderedAdaptiveCard renderedCard, String imageBaseUrl)
+    GenericImageLoaderAsync(RenderedAdaptiveCard renderedCard, String imageBaseUrl, int maxWidth)
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/GenericImageLoaderAsync.java]: > GenericImageLoaderAsync(RenderedAdaptiveCard renderedCard, String imageBaseUrl, int maxWidth) [](start = 3, length = 94)

keep the original constructor and add a flag or set m_maxWidth to -1 to know when was the maxWidth actually defined

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/InnerImageLoaderAsync.java [hunk 3]
```diff
@@ -9,9 +9,9 @@
 {
     protected View m_view; // button and imageview inherit from this
 
-    public InnerImageLoaderAsync(RenderedAdaptiveCard renderedCard, View containerView, String imageBaseUrl)
+    public InnerImageLoaderAsync(RenderedAdaptiveCard renderedCard, View containerView, String imageBaseUrl, int maxWidth)
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/InnerImageLoaderAsync.java]: >public InnerImageLoaderAsync(RenderedAdaptiveCard renderedCard, View containerView, String imageBaseUrl, int maxWidth) [](start = 4, length = 118)

keep the original constructor here too

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ImageRenderer.java [hunk 4]
```diff
@@ -58,9 +58,15 @@ public static ImageRenderer getInstance()
...
+        ImageRendererImageLoaderAsync(
+            RenderedAdaptiveCard renderedCard,
+            ImageView imageView,
+            String imageBaseUrl,
+            ImageStyle imageStyle,
+            int backgroundColor,
+            int maxWidth)
         {
-            super(renderedCard, imageView, imageBaseUrl);
+            super(renderedCard, imageView, imageBaseUrl, maxWidth);
```
Comment (almedina-ms) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ImageRenderer.java]: >super(renderedCard, imageView, imageBaseUrl, maxWidth); [](start = 12, length = 55)

keep both constructors here too

File: source/android/mobile/src/main/java/io/adaptivecards/adaptivecardssample/ImageUtil.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package io.adaptivecards.adaptivecardssample;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+
+public class ImageUtil {
+    public static Bitmap drawableToBitmap(Drawable drawable, int maxWidth) {
```
Comment (almedina-ms) [inline source/android/mobile/src/main/java/io/adaptivecards/adaptivecardssample/ImageUtil.java]: >{ [](start = 75, length = 1)

curly braces go on new lines

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/GenericImageLoaderAsync.java (hunk 2 above)
Comment (xinguohenan) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/GenericImageLoaderAsync.java]: Done.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/InnerImageLoaderAsync.java (hunk 3 above)
Comment (xinguohenan) [reply source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/InnerImageLoaderAsync.java]: Done.

**Pull request merged**